File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/data/db/dao/TaskDao.java
package com.vladislavmyasnikov.courseproject.data.db.dao;

import com.vladislavmyasnikov.courseproject.data.db.entity.TaskEntity;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

@Dao
public interface TaskDao {

    @Query("SELECT * FROM tasks WHERE lecture_id = :lectureId")
    LiveData<List<TaskEntity>> loadTasks(int lectureId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertTasks(List<TaskEntity> tasks);
}

File: Course-Project/app/src/main/java/com/vladislavmyasnikov/courseproject/data/db/dao/TaskDao.kt
package com.vladislavmyasnikov.courseproject.data.db.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.vladislavmyasnikov.courseproject.data.db.entities.TaskEntity

@Dao
interface TaskDao {

    @Query("SELECT * FROM tasks WHERE lecture_id = :id")
    fun loadTasks(id: Int): List<TaskEntity>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertTasks(tasks: List<TaskEntity>)
}
